src/client/src/components/pages/participants/participantDetails/index.jsx

```diff
@@ -29,31 +29,68 @@ class index extends Component {
   getDetails = async () => {
     const id = this.props.match.params.id;
     const { dispatch } = this.props.context;
-    axios(`/api/v2/participant/${id}`).then(result => {
-      const { data } = result;
-      const date = data.date_of_birth.split("T")[0];
-      this.setState({ ...data, date_of_birth: date });
-    }).catch(error => {
-      dispatch({ type: 'ERROR_PAGE', payload: { ErrorPage: error.response.status } })
-    })
+    axios(`/api/v2/participant/${id}`)
+      .then(result => {
+        const { data } = result;
+        const date = data.date_of_birth.split("T")[0];
+        this.setState({ ...data, date_of_birth: date });
+      })
+      .catch(error => {
+        dispatch({
+          type: "ERROR_PAGE",
+          payload: { ErrorPage: error.response.status }
+        });
+      });
+  };
 
+  updateParticipant = async obj => {
+    const confirm = await swal({
+      type: "warning",
+      html: "Are you sure that you want to update this data ?",
+      showCancelButton: true,
+      focusConfirm: false,
+      confirmButtonText: '<i class="fa fa-thumbs-up"></i> Yes',
+      confirmButtonAriaLabel: "Thumbs up",
+      cancelButtonText: '<i class="fa fa-thumbs-down"></i> No ',
+      cancelButtonAriaLabel: "Thumbs down"
+    });
+    if (confirm.value) {
+      const { id } = this.props.match.params;
+      const result = await axios(`/api/v2/participant/${id}`, {
+        method: "PUT",
+        data: {
+          participantData: obj
```

Comment:
This all looks good but I think there might be a bug with the date_of_birth field being sent back to the server in an unexpected format:
Up above, on line 35-36, you parse date_of_birth to get the year (so for example, the server gives you '1985-12-10T00:00:00Z" and you parse that to put '1985-12-10' into this.state.date_of_birth).
Then, you take this.state and PUT that to the endpoint, which means that you're PUTting that date_of_birth that you formatted into '1985-12-10', when the server probably needs the original format, '1985-12-10T00:00:00Z'.

I've done this bug a lot too, where I take server data, make small changes to it for displaying to the user, and then accidentally send the reformatted data, not the original format of the data, back to the server when saving it.  The fix is to always try to store the raw server data in this.state, and do that date formatting inside render() instead of inside getDetails().

Reply:
I'm not sure if it notifies you since I'm commenting on a merged PR so I'll just @ you, @mohannadhanafi